### ISSUE TITLE
Rework repo tester to just use cloning. [phab:D2313]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2778]
_Created by **danielmoy-google** at 2018-05-16 21:47:32_: 

Also support mirrors of github by not restricting url format.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2018-05-16 22:18:46_:

Simpler:  https://play.golang.org/p/gGeensP8L3B

This isn't really specific to a URL, this works for anything. Maybe a more generic name: `pathTail`?  

Separately: Do we need to worry about query parameters, fragments, or other non-path components?

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2018-05-16 22:18:46_:

`repo` → `repoURL` ? (I think that's what we're assuming)

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2018-05-16 22:18:46_:

Simpler:

```
	err = filepath.Walk(repoDir, func(path string, info os.FileInfo, err error) error {
		if err == nil && filepath.Ext(path) == ".java" {
			ret[path] = true
		}
		return err
	})
	return ret, err
```

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2018-05-16 22:18:46_:

`if err := exec.Command("git", "clone", repoURL, repoDir).Run(); err != nil {`

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2018-05-16 22:18:46_:

Completely optional nit: I suggest omitting periods at the end of flag strings, since they're not complete sentences (in most cases).

**Comment by kythe** [comment]
_**craigbarber** accepted this revision at 2018-05-16 22:30:16_

**Comment by kythe** [comment]
_**danielmoy-google** wrote at 2018-05-16 22:32:22_:

  - addressing comments

**Comment by kythe** [comment]
_**danielmoy-google** commented inline (context lost during migration) at 2018-05-16 22:32:32_:

presumably we do not need to worry about params/fragments, though if the time comes we can add support.  Previous logic I had here is some somewhat ugly regex (see above in prior version), so we can always go back to that.

**Comment by kythe** [comment]
_**danielmoy-google** wrote at 2018-05-16 22:32:32_:

thanks, addressed comments

**Comment by kythe** [comment]
_**danielmoy-google** wrote at 2018-05-16 22:35:18_:

  - use similar construction for other filepath.Walk

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2018-05-16 22:49:31_:

Yeah—if it comes to that, we can use http://godoc.org/net/url#Parse.

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2018-05-16 22:49:31_

**Comment by kythe** [comment]
_**danielmoy-google** closed via commit d08747cd9f3e31a156047793000119a3a7cf2580 at 2018-05-16 23:08:49_
